### PR TITLE
added searching and loading more teams features with tests

### PR DESCRIPTION
**Pull request by YuYing-Liang**
## Overview

- Copied over searching and loading more teams from newhacks repo

## Unit Tests Created

- made unit tests for searching teams, loading more teams and the teams search bar

## Steps to QA

- test searching teams, clearing search and loading more teams works properly on the hss admin side

### CODE REVIEW

File: hackathon_site/dashboard/frontend/src/pages/Teams/Teams.test.tsx [hunk 1]
```diff
@@ -47,4 +52,186 @@ describe("Teams Page", () => {
...
+        fireEvent.change(input, { target: { value: expectedFilters.search } });
+        fireEvent.submit(input);
+
+        await waitFor(() => {
+            expect(get).toHaveBeenCalledWith(teamsUri, expectedFilters);
+            expect(getByText("NO TEAMS FOUND")).toBeInTheDocument();
+        });
+    });
+
+    it("Loads more teams", async () => {
```
Comment (Leo6Leo) [inline hackathon_site/dashboard/frontend/src/pages/Teams/Teams.test.tsx:142]: We could have one more test: if the number of teams is within the limit or all teams have been loaded, the load more button should be displayed

File: hackathon_site/dashboard/frontend/src/components/team/TeamSearchBar/TeamSearchBar.test.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        const expectedFilters = {
+            search: "foobar",
+            limit: NUM_TEAM_LIMIT,
+        };
+
+        await waitFor(() => {
+            expect(get).toHaveBeenCalledWith(teamsUri, expectedFilters);
+        });
+    });
+
```
Comment (Leo6Leo) [inline hackathon_site/dashboard/frontend/src/components/team/TeamSearchBar/TeamSearchBar.test.tsx:30]: hitting "Enter" on the keyboard also work

File: hackathon_site/dashboard/frontend/src/pages/Teams/Teams.tsx [hunk 3]
```diff
@@ -8,24 +8,35 @@ import TeamCardAdmin from "components/team/TeamCardAdmin/TeamCardAdmin";
...
+    teamCountSelector,
 } from "slices/event/teamAdminSlice";
 import { useDispatch, useSelector } from "react-redux";
 import CircularProgress from "@material-ui/core/CircularProgress";
 import { useHistory } from "react-router-dom";
+import { Button, Divider } from "@material-ui/core";
 
 const Teams = () => {
     const dispatch = useDispatch();
     const teamsList = useSelector(teamAdminSelectors.selectAll);
```
Comment (Leo6Leo) [inline hackathon_site/dashboard/frontend/src/pages/Teams/Teams.tsx]: switch this with line 26, so that we put all `useSelector` together

**Review by Leo6Leo — CHANGES_REQUESTED** [review]
As commented above. The basic manual functionality test has passed.

File: hackathon_site/dashboard/frontend/src/pages/Teams/Teams.test.tsx (hunk 1 above)
Comment (YuYing-Liang) [reply hackathon_site/dashboard/frontend/src/pages/Teams/Teams.test.tsx:142]: Correct me if I'm wrong but I think that's what's tested in the following test (line 186)

File: hackathon_site/dashboard/frontend/src/components/team/TeamSearchBar/TeamSearchBar.test.tsx (hunk 2 above)
Comment (YuYing-Liang) [reply hackathon_site/dashboard/frontend/src/components/team/TeamSearchBar/TeamSearchBar.test.tsx:30]: done

File: hackathon_site/dashboard/frontend/src/pages/Teams/Teams.tsx (hunk 3 above)
Comment (YuYing-Liang) [reply hackathon_site/dashboard/frontend/src/pages/Teams/Teams.tsx]: done

**Review by Leo6Leo — DISMISSED** [review]
Looks good to me, great work Yuyingg!!!

**Review by Mustaballer — APPROVED** [review]

**Pull request merged**